src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
             return falseIfGuardFailed(e);
         }
     }
 
-    private async handleScreenEvent(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
-        const value = (input.request as interfaces.alexa.presentation.apl.UserEvent).arguments![1];
-
-        this.setValue(value);
-        this.state.isValueConfirmed = true;
-
-        const validationResult: true | ValidationFailure = await evaluateValidationProp(
-            this.props.validation,
-            this.state,
-            input,
-        );
-        if (validationResult !== true) {
-            await this.fixInvalidValue(input, resultBuilder);
-        } else {
-            this.state.isValidValue = true;
-            resultBuilder.addAct(
-                new ValueSetAct(this, { value, renderedValue: this.props.valueRenderer(value, input) }),
-            );
-        }
-        return;
-    }
-
-    private handleLastQuestionEmptyAndValueNotExisting(
-        input: ControlInput,
-        resultBuilder: ControlResultBuilder,
-    ): void {
-        resultBuilder.addAct(new RequestValueAct(this));
-    }
-
-    private async handleLastQuestionEmptyAndValueExisting(
+    /**
+     * Handle an implicit or explicit "set" with a value provided.
+     *
+     * @param input - Input
+     * @param resultBuilder - ResultBuilder
+     */
+    private async handleSetWithValue(
         input: ControlInput,
         resultBuilder: ControlResultBuilder,
     ): Promise<void> {
-        const { action, values } = unpackValueControlIntent((input.request as IntentRequest).intent);
-        const valueStr = values[0].slotValue;
+        const { valueStr } = InputUtil.getValueResolution(input);
         this.setValue(valueStr);
-        await this.commonHandlerWhenValueChanged(action ?? $.Action.Set, input, resultBuilder);
-    }
 
-    private canHandleForExistingStateValue(input: ControlInput): boolean {
-        try {
-            okIf(this.state.value !== undefined);
-            return (
-                this.isValueInRejectedValues(input) ||
-                this.isBareNoWhenConfirmingValue(input) ||
-                this.isFeedbackNoAndValueUndefinedWhenConfirmingValue(input) ||
-                this.isFeedbackNoAndValueNotChangedWhenConfirmingValue(input) ||
-                this.isFeedbackNoAndValueChangedWhenConfirmingValue(input) ||
-                this.isBareYesConfirmingValue(input) ||
-                this.isFeedbackYesAndValueChangedWhenConfirmingValue(input) ||
-                this.isFeedbackYesAndValueNotChangedWhenConfirmingValue(input) ||
-                this.isFeedbackYesAndValueUndefinedWhenConfirmingValue(input) ||
-                this.isFeedbackUndefinedAndValueNotChangedWhenConfirmingValue(input) ||
-                this.isFeedbackUndefinedAndValueChangedWhenConfirmingValue(input) ||
-                this.isTargetsMatchWithoutFeedbackNorValueWhenConfirmingValue(input)
+        if (this.isConfirmationRequired(input) === true) {
```

Comment:
It would be best if all initiative that can be delayed is managed through initiativeHandlers.  (there might be something higher-priority to discuss with the user).

Our tenet is that initiative is only produced during handle phase should be questions to resolve understanding of the input.

Reply:
Discussed offline, we agreed to keep the implementation of check for confirmations and valid input in handlers since delegating it to Initiative causes conflicts on order of initiative handlers since they won't be mutually exclusive
For eg `RequestReplacementForInvalidValue` and `ConfirmValue` both return True.

Also, for a simple dialog ordering of prompts changes too
U: set my age to 6 (invalid age)
A: Great, Sorry the input value must be even

U: set my age to 6 (invalid age)
A: Great (ValueSetAct from handlers), Was that 6?(ConfirmValue from Initiative Phase)